core/trino-main/src/test/java/io/trino/sql/planner/iterative/rule/TestMultipleDistinctAggregationsToSubqueries.java

```diff
@@ -0,0 +1,1694 @@
+                                                                tableScan(
+                                                                        TABLE_NAME,
+                                                                        ImmutableMap.of(
+                                                                                "input1_1_2Symbol", COLUMN_1,
+                                                                                "input2_1_2Symbol", COLUMN_2,
+                                                                                "right_groupingKey1", GROUPING_KEY_COLUMN))),
+                                                        filter(
+                                                                new Comparison(GREATER_THAN, new Reference(BIGINT, "input2_2_2Symbol"), new Constant(BIGINT, 2L)),
+                                                                tableScan(
+                                                                        TABLE_NAME,
+                                                                        ImmutableMap.of(
+                                                                                "input1_2_2Symbol", COLUMN_1,
+                                                                                "input2_2_2Symbol", COLUMN_2,
+                                                                                "right_groupingKey2", GROUPING_KEY_COLUMN))))
+                                                        .withAlias("input1Symbol2", new SetOperationOutputMatcher(0))
+                                                        .withAlias("input2Symbol2", new SetOperationOutputMatcher(1))
+                                                        .withAlias("right_groupingKey", new SetOperationOutputMatcher(2)))))));
+    }
+
+    private static TableHandle testTableHandle(RuleTester ruleTester)
+    {
+        return new TableHandle(ruleTester.getCurrentCatalogHandle(), new TpchTableHandle("sf1", TABLE_NAME, 1.0), TestingTransactionHandle.create());
+    }
+
+    private static RuleTester tester(boolean connectorSupportsSingleColumnReads)
+    {
+        Session session = testSessionBuilder().build();
+        PlanTester planTester = PlanTester.create(
+                session,
+                metadata -> new DelegatingMetadata(metadata)
+                {
+                    @Override
+                    public boolean isColumnarTableScan(Session session, TableHandle tableHandle)
+                    {
+                        return connectorSupportsSingleColumnReads;
+                    }
+                },
+                1);
+        planTester.createCatalog(
+                session.getCatalog().orElseThrow(),
+                new TpchConnectorFactory(1),
+                ImmutableMap.of());
+        return new RuleTester(planTester);
+    }
+
+    private static class DelegatingMetadata
+            implements Metadata
```

Comment:
Can we just extend `AbstractMockMetadata` and avoid implementing entire interface ?